tests/unit/test_charm.py

```diff
@@ -4,64 +4,39 @@
 # Learn more about testing at: https://juju.is/docs/sdk/testing
 
 import unittest
-from unittest.mock import Mock
+from unittest.mock import patch
 
-from ops.model import ActiveStatus
+from ops.model import ActiveStatus, WaitingStatus
 from ops.testing import Harness
 
-from charm import OperatorTemplateCharm
+from charm import MysqlOperatorCharm
 
 
 class TestCharm(unittest.TestCase):
     def setUp(self):
-        self.harness = Harness(OperatorTemplateCharm)
+        self.harness = Harness(MysqlOperatorCharm)
         self.addCleanup(self.harness.cleanup)
         self.harness.begin()
 
-    def test_config_changed(self):
-        self.assertEqual(list(self.harness.charm._stored.things), [])
-        self.harness.update_config({"thing": "foo"})
-        self.assertEqual(list(self.harness.charm._stored.things), ["foo"])
-
-    def test_action(self):
-        # the harness doesn't (yet!) help much with actions themselves
-        action_event = Mock(params={"fail": ""})
-        self.harness.charm._on_fortune_action(action_event)
-
-        self.assertTrue(action_event.set_results.called)
-
-    def test_action_fail(self):
-        action_event = Mock(params={"fail": "fail this"})
-        self.harness.charm._on_fortune_action(action_event)
-
-        self.assertEqual(action_event.fail.call_args, [("fail this",)])
-
-    def test_httpbin_pebble_ready(self):
-        # Check the initial Pebble plan is empty
-        initial_plan = self.harness.get_container_pebble_plan("httpbin")
-        self.assertEqual(initial_plan.to_yaml(), "{}\n")
-        # Expected plan after Pebble ready with default config
-        expected_plan = {
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {"thing": "🎁"},
-                }
-            },
-        }
-        # Get the httpbin container from the model
-        container = self.harness.model.unit.get_container("httpbin")
-        # Emit the PebbleReadyEvent carrying the httpbin container
-        self.harness.charm.on.httpbin_pebble_ready.emit(container)
-        # Get the plan now we've run PebbleReady
-        updated_plan = self.harness.get_container_pebble_plan("httpbin").to_dict()
-        # Check we've got the plan we expected
-        self.assertEqual(expected_plan, updated_plan)
-        # Check the service was started
-        service = self.harness.model.unit.get_container("httpbin").get_service("httpbin")
-        self.assertTrue(service.is_running())
-        # Ensure we set an ActiveStatus with no message
-        self.assertEqual(self.harness.model.unit.status, ActiveStatus())
+    def test_on_pebble_ready(self):
```

Comment:
You'll want to [this scenario](https://github.com/canonical/mysql-k8s-operator/blob/4212fd595b7738a8a7f2dcada5d7e8a59e0eaa65/src/charm.py#L59-L62) from `charm.py` specifically when `current_layer.services != new_layer.services` and `current_layer.services == new_layer.services`

Reply:
I'm sorry, I don't get your comment, could you please elaborate?

Reply:
Ah I had a typo in it so it didn't make sense, I corrected it but let me try to be more clear 😅 

I was trying to say that the current test `test_on_pebble_ready ` tests [this part of the code](https://github.com/canonical/mysql-k8s-operator/blob/4212fd595b7738a8a7f2dcada5d7e8a59e0eaa65/src/charm.py#L45-L57) and the Active Status part. But I don't think it tests the case where the `layer.services` differ, ie [this part](https://github.com/canonical/mysql-k8s-operator/blob/4212fd595b7738a8a7f2dcada5d7e8a59e0eaa65/src/charm.py#L59-L63) and that you might want to test it to be sure it adds a layer when necessary.

Reply:
Ah, I get it. Although we have to, I'm not testing the expected `Layer` vs the actual layer (like in Postgres, even in Mongo I think) at least in this PR. I left it like that because I'm pretty sure I'll change the layer definitions, and I didn't want to add a test that I'll have to change later. I'm planning to do something similar to what @marceloneppel did for [Postgres](https://github.com/canonical/postgresql-k8s-operator/blob/main/tests/unit/test_charm.py#L78).